### DISCUSSION TITLE
Possible to to GET request with body?

### DISCUSSION THREAD

**Discussion by boyter** [earthboundkid/requests#66] [Support] [answered]
For reasons probably best not answered I am in the un-enviable position of needing to make GET requests to a 3rd party API (which I have no control over) with a JSON BODY being present.

Is there some way to override the default behavior which would cause this to be a POST? I had a look through the source and didn't see one immediately, but that could just be me being stupid.

Not related to the above, but I love this library. Pushes the happy parts of my brain the same way that Python Requests does. Thanks so much for creating it.

**Comment by earthboundkid** [comment, marked as answer]
Thanks!

If you set the Method that should take precedence, even if you have a body.

**Reply by boyter** [reply]
Ah so it does. I must have made a mistake when I first did this and couldn't find anything. Hopefully this helps someone else in the future.
